=== modified file 'setup.py'
--- setup.py	2011-08-04 14:52:04 +0000
+++ setup.py	2013-01-11 11:37:52 +0000
@@ -27,6 +27,9 @@
       author_email='brywilharris@gmail.com',
       license='GPL v3 or later',
       url='https://launchpad.net/reduce',
-      long_description='A program for making pretty pictures of a whole directory test data at once.'
-      )
+      long_description='A program for making pretty pictures of a whole directory test data at once.',
+      options = {'py2exe':{'bundle_files':1}},
+      console = ['rlo.py'],
+      zipfile = None,
+)
 

