diff -Naur reduce-2.2.1-jep06/HISTORY reduce-2.2.1-jep07/HISTORY
--- reduce-2.2.1-jep06/HISTORY	2011-07-22 19:14:01.201828442 -0400
+++ reduce-2.2.1-jep07/HISTORY	2011-07-22 20:12:58.861354038 -0400
@@ -1,3 +1,18 @@
+2011-07-22 reduce-2.2.1-jep07
+
+    rlo.py
+
+        Redirected almost all normal print output to both 
+        standard output and the log file. Refactored such code. 
+        Fixed probable bug that had printed onto two lines what 
+        was likely supposed to be on a single line. (Lines 212-214). 
+
+        Added printing of some blank lines between major sections 
+        of activity. 
+
+    New code tries to follow PEP 8 -- Style Guide for Python Code
+    http://www.python.org/dev/peps/pep-0008/
+
 2011-07-22 reduce-2.2.1-jep06
 
     rlo.py and dataFile.py
diff -Naur reduce-2.2.1-jep06/rlo.py reduce-2.2.1-jep07/rlo.py
--- reduce-2.2.1-jep06/rlo.py	2011-07-22 19:15:21.920813666 -0400
+++ reduce-2.2.1-jep07/rlo.py	2011-07-22 19:57:56.380699757 -0400
@@ -73,16 +73,21 @@
 # where am I?
 base_path = os.getcwd() 
 
+# Writing to t writes all its listed files.  
+t=Tee([
+    sys.stdout, 
+    open(os.path.join(base_path, 'script.log'), 'w')])
+
 # Here I am.
-print "base_path: " + base_path 
+print >>t, 'base_path:', base_path 
 
 # Where to put reduced data
 reduce_path=os.path.join(base_path,"reduced")  
-print "reduce_path: " + reduce_path  
+print >>t, 'reduce_path:', reduce_path  
 
 # Where to put pretty pictures
 image_path=os.path.join(reduce_path,"png") 
-print "image_path: " + image_path
+print >>t, 'image_path:', image_path
 
 # Create directories as needed. 
 for directory in [reduce_path, image_path]:
@@ -95,7 +100,7 @@
                 % directory)
             sys.exit(os.EX_USAGE)
         else:
-            print directory + ' created'
+            print >>t, directory, 'created'
 
     # Test for ability to write to directory. 
     if not os.access(directory, os.W_OK):
@@ -104,15 +109,13 @@
         sys.exit(os.EX_USAGE)
 
 # Where to generate measurement files
-log_file_name = "script.log"
-log_file_path = os.path.join(base_path,log_file_name)
 measurement_file_present=False 
 meas_file_name = "measurements.csv"  
 meas_file_path = os.path.join(base_path,meas_file_name)
-print "meas_file_path: " + meas_file_path
+print >>t, 'meas_file_path:', meas_file_path
 
 summary_file_path = os.path.join(reduce_path, 'summary.dat')
-print 'summary_file_path:', summary_file_path
+print >>t, 'summary_file_path:', summary_file_path
 summary_file = open(summary_file_path, 'w')
 print >>summary_file, 'Specimen\tRate\tPeak Load'
 
@@ -173,16 +176,16 @@
 # go into the working file directory and create the summary file
 
 os.chdir(reduce_path)
-log_file=open(log_file_path, 'w');
-t=Tee([sys.stdout, log_file])
 
 #Start looking at the data files
 
 print >>t, time.strftime('%a, %d %b %Y %H:%M:%S +0000', time.localtime())
 print >>t, 'user:', os.times()[0], 's'
 print >>t, 'system:', os.times()[1], ' s' 
+print >>t
 
 print >>t, 'Measurement Files:'
+print >>t
 
 for meas_file_name in measurement_file_names :
     print >>t, meas_file_name
@@ -207,8 +210,9 @@
         for specimen in mfn.specimens :
             #print specimen
             if foo_name.find(specimen.stl_ID) != -1:
-                print foo_name, "appears to have a measurement file entry:"
-                print specimen
+                print >>t, foo_name,
+                print >>t, 'appears to have a measurement file entry:',
+                print >>t, specimen
                 thisDataFile.Has_Measurement_File = True
                 thisDataFile.Thickness_in = specimen.thick
                 thisDataFile.Width_in = specimen.width
@@ -364,9 +368,9 @@
 
             slope = (load_data[slope_end] - load_data[slope_start])/\
                     (new_column[slope_end] - new_column[slope_start])
-            print "modulus = ", str(slope/thisDataFile.CS_Area_in2)
+            print >>t, 'modulus =', slope/thisDataFile.CS_Area_in2
             intercept = new_column[slope_start]-load_data[slope_start]/slope
-            print "intercept = ",str(intercept)
+            print >>t, 'intercept =', intercept
 
 
             for i in range(len(new_column)):
@@ -396,8 +400,7 @@
     print >>t, time.strftime('%a, %d %b %Y %H:%M:%S +0000', time.localtime())
     print >>t, 'user:', os.times()[0], 's'
     print >>t, 'system:', os.times()[1], 's'
-
-    print >>t, ''
+    print >>t
     
 
     #(end of analysis loop)
@@ -406,5 +409,4 @@
 #Were all done, close the summary file
 
 summary_file.close()
-log_file.close()
 
