diff -Naur reduce-2.2.3-jep04/HISTORY reduce-2.2.3-jep05/HISTORY
--- reduce-2.2.3-jep04/HISTORY	2011-07-28 10:59:02.427425704 -0400
+++ reduce-2.2.3-jep05/HISTORY	2011-07-28 17:21:35.134871095 -0400
@@ -1,3 +1,9 @@
+2011-07-28 reduce-2.2.3-jep05 James Prior
+
+    reducePlot.py
+
+        Named a magic number: MAX_PLOT_POINTS = 500
+
 2011-07-28 reduce-2.2.3-jep04 James Prior
 
     dataFile.py
diff -Naur reduce-2.2.3-jep04/MD5SUM reduce-2.2.3-jep05/MD5SUM
--- reduce-2.2.3-jep04/MD5SUM	2011-07-28 10:59:07.743358874 -0400
+++ reduce-2.2.3-jep05/MD5SUM	2011-07-28 17:23:27.049464137 -0400
@@ -1,8 +1,8 @@
 d22d26f5e12df29d258779afb9c24321  AUTHORS
 6db9e7a3fdd32ff07ee2dadc1bed6148  dataFile.py
-be060c78b64c168fe3d3b47f9306ce74  HISTORY
+3df4a8ad0dfd6862fca69462451f5893  HISTORY
 7eb888b27c8dfbfe88545385dc12a12b  measurementFile.py
-bf068a6c2fd18e81fb08e0cbfba4e60e  reducePlot.py
+906d1d396a15b7f49fef8bf4db791c58  reducePlot.py
 17e163e9d29bfa64d4aa8c2e6dc8fea1  rlo.py
 92f9899ed684652be377a18488d867da  setup.py
 fa1160c24b07502d528f172c09902ab4  tee.py
diff -Naur reduce-2.2.3-jep04/reducePlot.py reduce-2.2.3-jep05/reducePlot.py
--- reduce-2.2.3-jep04/reducePlot.py	2011-07-27 20:57:41.000000000 -0400
+++ reduce-2.2.3-jep05/reducePlot.py	2011-07-28 17:23:16.285599458 -0400
@@ -21,6 +21,8 @@
 import tempfile
 import os
 
+MAX_PLOT_POINTS = 500
+
 def decimate(array, points):
     if not isinstance(points, int):
         raise Error, 'number of points should be an integer'
@@ -113,8 +115,8 @@
 
     picturename = f.filebase + '-rate_measure.png'
     p.title('Time - Disp - Load' + '\n' + os.path.basename(f.filebase))
-    p.plot(decimate(time_trace[start_line:end_line], 500), \
-        decimate(stroke_trace[start_line:end_line], 500), \
+    p.plot(decimate(time_trace[start_line:end_line], MAX_PLOT_POINTS),
+        decimate(stroke_trace[start_line:end_line], MAX_PLOT_POINTS),
         'b-', label='Disp')
 
     stroke_rate = f.find_rate(start_line, end_line)
@@ -131,8 +133,8 @@
     p.legend(loc=2)
 
     p.twinx()
-    p.plot(decimate(time_trace[start_line:end_line], 500),
-        decimate(f.get_zl_data()[start_line:end_line], 500),
+    p.plot(decimate(time_trace[start_line:end_line], MAX_PLOT_POINTS),
+        decimate(f.get_zl_data()[start_line:end_line], MAX_PLOT_POINTS),
         'r--', label='Load')
     v = p.axis()
     p.axis((v[0], v[1], 0, v[3]))
