diff -Naur reduce-2.2.3-jep01/HISTORY reduce-2.2.3-jep02/HISTORY
--- reduce-2.2.3-jep01/HISTORY	2011-07-27 23:52:41.478054307 -0400
+++ reduce-2.2.3-jep02/HISTORY	2011-07-28 10:29:19.133844741 -0400
@@ -1,3 +1,9 @@
+2011-07-28 reduce-2.2.3-jep02 James Prior
+
+    tee.py
+
+        Slightly improved docstring for Tee class. 
+
 2011-07-27 reduce-2.2.3-jep01 James Prior
 
     measurementFile.py
diff -Naur reduce-2.2.3-jep01/MD5SUM reduce-2.2.3-jep02/MD5SUM
--- reduce-2.2.3-jep01/MD5SUM	2011-07-27 23:52:47.017984658 -0400
+++ reduce-2.2.3-jep02/MD5SUM	2011-07-28 10:30:29.300962621 -0400
@@ -1,8 +1,8 @@
 d22d26f5e12df29d258779afb9c24321  AUTHORS
 0db3e389e3336bc64d0122f81821baf9  dataFile.py
-0fb8c8be97da8f666dc68894f267894e  HISTORY
+bdec1ac9db987a2abc47a3bcd56de6db  HISTORY
 da994c0896f4c5e923b850cb23167f09  measurementFile.py
 bf068a6c2fd18e81fb08e0cbfba4e60e  reducePlot.py
 17e163e9d29bfa64d4aa8c2e6dc8fea1  rlo.py
 92f9899ed684652be377a18488d867da  setup.py
-46d4ed4205f776a25f2df8914261bdbd  tee.py
+fa1160c24b07502d528f172c09902ab4  tee.py
diff -Naur reduce-2.2.3-jep01/tee.py reduce-2.2.3-jep02/tee.py
--- reduce-2.2.3-jep01/tee.py	2011-07-27 20:57:41.000000000 -0400
+++ reduce-2.2.3-jep02/tee.py	2011-07-27 16:00:20.647256759 -0400
@@ -25,16 +25,21 @@
     one uses a single print that redirects to an instance of this class.  
 
     For example, instead of doing: 
+
         print 'big long ugly messy stuff'
         print >>file1 'big long ugly messy stuff'
         print >>file2 'big long ugly messy stuff'
-    one does the following:
+
+    one can do the following:
+
         import sys
         from tee import *
         tee=Tee([sys.stdout,file1,file2])
         ...
         print >>tee 'big long ugly messy stuff'
+
     or even:
+
         import sys
         from tee import *
         real_stdout = sys.stdout
